Sources/TensorFlow/Core/Utilities.swift

```diff
@@ -167,3 +167,153 @@ extension UnsafeMutablePointer where Pointee == CTensorHandle? {
         self.init(unwrapped)
     }
 }
+
+//===------------------------------------------------------------------------------------------===//
+// Hashing
+//===------------------------------------------------------------------------------------------===//
+
+internal extension FixedWidthInteger {
+    init(bytes: ArraySlice<UInt8>, startingAt index: Int) {
+        if bytes.isEmpty { self.init(0); return }
+        let count = bytes.count
+        self.init(0)
+        for i in 0..<MemoryLayout<Self>.size {
+            let j = (MemoryLayout<Self>.size - i - 1) * 8
+            self |= count > 0 ? Self(bytes[index.advanced(by: i)]) << j : 0
+        }
+    }
+
+    func bytes(count byteCount: Int = MemoryLayout<Self>.size) -> [UInt8] {
+        var littleEndianValue = littleEndian
+        return withUnsafePointer(to: &littleEndianValue) { pointer -> [UInt8] in
+            let bytesPointer = UnsafeMutablePointer<UInt8>(OpaquePointer(pointer))
```

Comment:
`OpaquePointer` is for interoperability with opaque C types. This is not the best way of turning some value into bytes. Instead, it would be highly preferable to use `UnsafeBufferPointer` (which is a `Collection`) and `UnsafeBufferPointer.withMemoryRebound(to:_:)`. Then, you can initialize an array directly without doing pointer arithmetics.

Reply:
Thanks for the suggestion! I'll fix that soon, after I open the PR for conditional differentiable conformance.

Reply:
Addressed in #337.